core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2736,7 +2736,7 @@ private[spark] class DAGScheduler(
     // If the RDD has some placement preferences (as is the case for input RDDs), get those
     val rddPrefs = rdd.preferredLocations(rdd.partitions(partition)).toList
     if (rddPrefs.nonEmpty) {
-      return rddPrefs.map(TaskLocation(_))
+      return rddPrefs.map(TaskLocation(_)).filter(_ != null)
```

Comment:
If you filter out null first, you don't need to check in the apply() method below

Reply:
sgtm